core/bindings.rs

```diff
@@ -453,17 +461,56 @@ fn opcall_async<'s>(
       state.pending_ops.push(fut);
       state.have_unpolled_ops = true;
     }
-    Op::AsyncUnref(fut) => {
-      state.op_state.borrow().tracker.track_unref(op_id);
-      state.pending_unref_ops.push(fut);
-      state.have_unpolled_ops = true;
-    }
     Op::NotFound => {
       throw_type_error(scope, format!("Unknown op id: {}", op_id));
     }
   }
 }
 
+fn ref_op<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  args: v8::FunctionCallbackArguments,
+  _rv: v8::ReturnValue,
+) {
+  let state_rc = JsRuntime::state(scope);
+  let mut state = state_rc.borrow_mut();
+
+  let promise_id = match v8::Local::<v8::Integer>::try_from(args.get(0))
```

Comment:
Minor caveat: the way try_from() is currently implemented for v8::Integer means it only works for numbers in the range -2**31..2**32. I suppose that's fine (I see you're using i32 elsewhere) but I'd be remiss if I didn't point it out.

`Value::integer_value()` and `Integer::value()` return i64s so you could be forgiven for assuming try_from() works for all numbers that fit in an i64, but unfortunately it currently doesn't.

edit: opened https://github.com/denoland/rusty_v8/issues/839

Reply:
I suggest to handle that in a follow up PR; we got a bunch of places using the same conversion.